src/Microsoft.Extensions.PlatformAbstractions/Native/NativeMethods.Darwin.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Extensions.PlatformAbstractions.Native
+{
+    internal static partial class NativeMethods
+    {
+        public static class Darwin
+        {
+            private const int CTL_KERN = 1;
+            private const int KERN_OSRELEASE = 2;
+
+            public unsafe static string GetKernelRelease()
+            {
+                // This is way to small. Using it to test errno/GetLastError
+                const uint BUFFER_LENGTH = 32;
+
+                var name = stackalloc int[2];
+                name[0] = CTL_KERN;
+                name[1] = KERN_OSRELEASE;
+
+                var buf = stackalloc byte[(int)BUFFER_LENGTH];
+                try
+                {
+                    var len = stackalloc uint[1];
+                    *len = BUFFER_LENGTH;
+                    if (sysctl(name, 2, buf, len, IntPtr.Zero, 0) != 0)
+                    {
+                        throw new PlatformNotSupportedException("Failed to get Darwin Kernel Version");
+                    }
+                    else if(*len >= BUFFER_LENGTH)
```

Comment:
Is this check needed at all actually? It looks to me that if the buffer is not big enough the call to `sysctl` will return a non zero return value so this condition is never reachable. If this is the case it would also be easy to change the code to not catch and rethrow the exception that was thrown just a few lines above.